component/xload/blockpool.go

```diff
@@ -97,15 +99,44 @@ func (pool *BlockPool) GetBlockSize() uint64 {
 	return pool.blockSize
 }
 
-// Get a Block from the pool, return back if nothing is available
-func (pool *BlockPool) Get() *Block {
+func (pool *BlockPool) GetBlock(priority bool) *Block {
+	if priority {
+		return pool.mustGet()
+	} else {
+		return pool.tryGet()
+	}
+}
+
+// TryGet a block from the pool. If the pool is empty, wait till a block is released back to the pool
+func (pool *BlockPool) tryGet() *Block {
 	// getting a block from pool will be a blocking operation if the pool is empty
 	b := <-pool.blocksCh
 
 	// Mark the buffer ready for reuse now
-	if b != nil {
-		b.ReUse()
+	b.ReUse()
+	return b
+}
+
+func (pool *BlockPool) mustGet() *Block {
+	var b *Block = nil
+
+	select {
+	case b = <-pool.blocksCh:
+		break
+
+	default:
```

Comment:
If we allocate a new block here then release shall also check if queue is full or not, and if so it shall release the memory otherwise this one block will linger for the lifespan of blobfuse.

Reply:
Already doing that, https://github.com/Azure/azure-storage-fuse/blob/sourav/prefetchOpen/component/xload/blockpool.go#L144-L151
The release of block back to the pool is non-blocking and if the pool is full, the block will be deleted, and its memory will be released.

```
func (pool *BlockPool) Release(b *Block) {
	select {
	case pool.blocksCh <- b:
		break
	default:
		_ = b.Delete()
	}
}
```